### PR TITLE
Don't coalesce tables to get size or to build preview columns

### LINKED ISSUE #3748: Exporting Table Causes it to Be Coalesced
**Description**
Running "x = ..." with a command that produces an Uncoalesced table would not coalesce the table.

**Expected results**

Viewing an uncoalesced table does not force coalescing.

**Versions**
 - Deephaven: 0.21.0

### PR DESCRIPTION
**Pull request by niloc132**
Fixes #3748

### CODE REVIEW

**Comment by niloc132** [comment]
To test this, I ran this python:
```python
from deephaven import empty_table
t1 = empty_table(10).update_view(['I=i', 'Partition=0'])
t2 = empty_table(10).update_view(['I=i', 'Partition=1'])
from deephaven.parquet import write
write(t1, '/tmp/table/P=p0/table.parquet')
write(t2, '/tmp/table/P=p1/table.parquet')
from deephaven.parquet import read
t = read('/tmp/table')
```

I put a breakpoint in UncoalescedTable.coalesce() to confirm it was only called as expected - when the UI tries to pick a partition, it always results in a call to coalesce(), but that appears to be how the individual partition is read.

File: proto/proto-backplane-grpc/src/main/proto/deephaven/proto/table.proto [hunk 1]
```diff
@@ -261,7 +261,9 @@ message ExportedTableCreationResponse {
   // Whether or not this table might change.
   bool is_static = 5;
 
-  // The current number of rows for this table.
+  // The current number of rows for this table. If this is negative, the table isn't coalesced, meaning the
+  // size isn't known without scanning partitions. Typically, the client should filter the data by the partitioned
```
Comment (rcaudy) [inline proto/proto-backplane-grpc/src/main/proto/deephaven/proto/table.proto:265]:
```suggestion
  // size isn't known without scanning partitions. Typically, the client should filter the data by the partitioning
```

**Review by rcaudy — APPROVED** [review]
Good enough for me. One suggested comment change in the .proto, but meh.

**Review by nbauernfeind — APPROVED** [review]
LGTM

**Pull request merged**